From 1f6b3bc02eec5007cb80fd5c3f7f11bf36755118 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 23 Mar 2022 09:31:05 +0200
Subject: [PATCH] [SW-76678] gaudi3: add boot termination config

In pcie recovery flow, it is expected to configure boot termination
registers else the inbound traffic will get blocked. This patch
add boot termination configuration which will be done for master die.

Change-Id: Ib5da3e7e94bf083be4a32f93fb2064cd79592784
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 413ef32..592c246 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -302,6 +302,12 @@
 	regmap_writel(id, mmD1_PCIE_DBI_SIG_DEVICE_ID_VENDOR_ID_REG);
 }
 
+void pcie_config_boot_termination(void)
+{
+	regmap_writel(D0_PCIE_AUX_BOOT_TERM_EN_VAL_M, mmD0_PCIE_AUX_BOOT_TERM_EN);
+	regmap_writel(D0_PCIE_AUX_BOOT_TERM_CLR_CLR_M, mmD0_PCIE_AUX_BOOT_TERM_CLR);
+}
+
 static void arcboot_rom_prstn_init(void)
 {
 	if (arcboot_rom_global.die_id) {
diff --git a/common/pcie.c b/common/pcie.c
index 8a994a7..d2cb969 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -72,6 +72,7 @@
 
 __weak void pcie_phy_ssc_en_set(void) {}
 __weak void pcie_set_pcie_device_vendor_id(void) {}
+__weak void pcie_config_boot_termination(void) {}
 
 void pcie_config(void)
 {
@@ -120,8 +121,10 @@
 		}
 
 		/* Configure IATU */
-		if (!arcboot_rom_global.die_id)
+		if (!arcboot_rom_global.die_id) {
 			pcie_config_iatu();
+			pcie_config_boot_termination();
+		}
 
 		/* enable flr handling by HW state machine */
 		enable_flr_handling_fsm();
